lib/fork.js

```diff
@@ -109,15 +150,24 @@ module.exports = (file, options, execArgv = process.execArgv) => {
 			resolve();
 		};
 
-		subprocess.on('message', message => {
+		port.on('message', message => {
 			if (!message.ava) {
 				return;
 			}
 
 			switch (message.ava.type) {
-				case 'ready-for-options':
-					send({type: 'options', options});
+				case 'ready-for-options': {
+					let options_ = options;
+					if (options.workerThreads) {
+						// Removing `providers` field because functions cannot be transfered in MessageChannel
+						const {providers, ...cleaned} = options;
+						options_ = cleaned;
+					}
```

Comment:
I don't mind keeping it here. For now, it is easy to find. I don't know where to move this logic atm. Do you have any suggestions?. I can surely remove the check on `workerThreads` since it is always required.

Reply:
Note: maybe it's my editor that named `options_`, I don't usually use that pattern. I'll check it.

Reply:
The unique `options` object is passed to `createWorker` so it could be serialized / cleaned before the thread / child process is even started.